game/scripts/npc/abilities/siltbreaker/siltbreaker_boss_protection.txt

```diff
@@ -120,10 +115,28 @@
           }
         }
 
+        // The actual protection is done on stun_protection.lua filter
         "States"
         {
           "MODIFIER_STATE_SILENCED"  "MODIFIER_STATE_VALUE_DISABLED"
           "MODIFIER_STATE_STUNNED"  "MODIFIER_STATE_VALUE_DISABLED"
+          "MODIFIER_STATE_HEXED"  "MODIFIER_STATE_VALUE_DISABLED"
```

Comment:
Have you tried making this a Lua modifier? It's entirely possible the priority doesn't work quite right in DataDriven.